### PR TITLE
Fixes for string; domain + JS identifier api support in script tags

### PR DESCRIPTION
**Pull request by dbMundada**

### CODE REVIEW

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/sink/JSAPITagger.scala [hunk 1]
```diff
@@ -100,6 +101,18 @@ class JSAPITagger(cpg: Cpg, ruleCache: RuleCache, privadoInput: PrivadoInput)
           addRuleTags(builder, scriptTag, ruleInfo, ruleCache, Some(newRuleIdToUse))
         }
         storeForTag(builder, scriptTag, ruleCache)(Constants.apiUrl + newRuleIdToUse, domain._1)
+      } else {
+        val identifierDomain =
+          getAPIIdentifierFromCode(scriptTag.code, identifierRegex).getOrElse(Constants.UnknownDomain)
+        if (!identifierDomain.equals(Constants.UnknownDomain)) {
+          if (ruleInfo.id.equals(Constants.internalAPIRuleId)) addRuleTags(builder, scriptTag, ruleInfo, ruleCache)
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/javascript/tagger/sink/JSAPITagger.scala:108]: I guess this should be not interanlAPIRuleId

File: src/main/scala/ai/privado/utility/Utilities.scala [hunk 2]
```diff
@@ -395,6 +396,12 @@ object Utilities {
     }
   }
 
+  def getAPIIdentifierFromCode(codeSnippet: String, variablePattern: String): Option[String] = {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/utility/Utilities.scala:399]: @dbMundada Can we have some test case around this method?

Comment (dbMundada) [reply src/main/scala/ai/privado/utility/Utilities.scala:399]: added tests

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**